tslearn/neighbors.py

```diff
@@ -64,37 +123,33 @@ def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
         if self.metric == "precomputed":
             full_dist_matrix = X
         else:
-            parallelize = False
-            if self.metric == "dtw" or self.metric == cdist_dtw:
-                cdist_fun = cdist_dtw
-                parallelize = True
-            elif self.metric == "softdtw" or self.metric == cdist_soft_dtw:
-                cdist_fun = cdist_soft_dtw
-            elif self.metric in ["euclidean", "sqeuclidean", "cityblock"]:
-                def cdist_fun(X, Xp):
-                    return scipy_cdist(X.reshape((X.shape[0], -1)),
-                                       Xp.reshape((Xp.shape[0], -1)),
-                                       metric=self.metric)
-            else:
-                raise ValueError("Unrecognized time series metric string: %s "
-                                 "(should be one of 'dtw', 'softdtw', "
-                                 "'euclidean', 'sqeuclidean' "
-                                 "or 'cityblock')" % self.metric)
 
             if X.ndim == 2:  # sklearn-format case
                 X = X.reshape((X.shape[0], -1, self._d))
                 fit_X = self._X_fit.reshape((self._X_fit.shape[0],
                                              -1,
                                              self._d))
+            elif hasattr(self, '_ts_fit') and self._ts_fit is not None:
+                fit_X = self._ts_fit
             else:
                 fit_X = self._X_fit
-            if parallelize:
-                full_dist_matrix = cdist_fun(X, fit_X, n_jobs=self.n_jobs,
-                                             verbose=self.verbose, 
-                                             **metric_params)
+
+            if (self.metric in VARIABLE_LENGTH_METRICS or
```

Comment:
The test should not be this one, it should be something like "is it a tslearn metric?" maybe we could define such a list in the header of `neighbors.py`, something like:

```python
TSLEARN_VALID_METRICS = ['dtw', 'softdtw', 'sax']
neighbors.VALID_METRICS['brute'].extend(TSLEARN_VALID_METRICS)
```

Reply:
VARIABLE_LENGTH_METRICS is already defined in the header of tslearn/metrics.py, so I think this does check whether it is a valid tslearn metric?

Reply:
Yes, but I meant: at this point, we want to add `tslearn` metrics to kNN class, even if these functions do not operate on variable-length data. So, in terms of semantics, what we want to do is add tslearn-specific metrics, not necessarily variable-length metrics. 

Not sure if I'm clear here.

Reply:
Ok makes sense. So do you want to keep track of both where the variable-length one is a subset of all tslearn metrics?

Reply:
Yes, something like that would be great.

Reply:
Did you mean something like this? (see latest commit)